app/common/src/services/Backend.ts

```diff
@@ -823,11 +818,19 @@ export function createRootDirectoryAsset(directoryId: DirectoryId): DirectoryAss
     parentId: DirectoryId(''),
     permissions: [],
     projectState: null,
+    extension: null,
     labels: [],
     description: null,
+    parentsPath: '',
+    virtualParentsPath: '',
   }
 }
 
+/** Extract the file extension from a file name. */
+function fileExtension(fileNameOrPath: string) {
+  return fileNameOrPath.match(/[.]([^.]+?)$/)?.[1] ?? ''
+}
+
```

Comment:
There's also [path.extname](https://www.w3schools.com/nodejs/met_path_extname.asp), however I'm not sure if available here.

Reply:
indeed not. we could theoretically polyfill node's builtin modules, but i guess i'd rather not